usr/lib/linuxmuster-webui/plugins/lmn_linbo_sync/api.py

```diff
@@ -183,19 +184,51 @@ def test_online(host):
 
     :param host: Hostname
     :type host: string
-    :return: OS type
+    :return: OS type (Off, Linbo, OS Linux, OS Windows, OS Unknown)
     :rtype: string
     """
-
-    command=['nmap', '-p', '22', '-O', host]
+    command=["nmap", "-p", "2222,22,135", host, "-oX", "-"]
     r = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=False).stdout.read()
-    if b'Too many fingerprints' in r:
-        return "Linbo"
-    elif b'Linux' in r:
+    xmlRoot = ET.fromstring(r)
+    openPorts = []
+
+    try:
+        hostElement = xmlRoot.findall("host")[0]
+        portsElement = hostElement.findall("ports")[0]
+        scannedPorts = portsElement.findall("port")
+    except:
+        return get_os_from_open_ports([])
+
+    for scannedPort in scannedPorts:
+        portNumber = scannedPort.attrib["portid"]
+        portState = scannedPort.findall("state")[0].attrib["state"]
+        if portState == "open":
```

Comment:
This will not work everey time, because some ports may appear as "filtered".

Reply:
So, should I also accept `filtered`?

Reply:
Yes : `if portState in [...]`

Reply:
I need to rethink the logic a bit in that case. Because Windows looks like this:
![image](https://user-images.githubusercontent.com/30153207/148693819-f0471cc8-e57c-4b16-92d5-73f436275267.png)

What exactly does filtered mean? Are you sure that it is also valid to identify an open port?

Reply:
It does not matter. I'm going to do some more testing and clean it up.

Reply:
`filtered` means that this port is mostly protected by a firewall, and `nmap` cannot do further test to determine what's going on. I got a lot of filtered answers on my network, that's why this command last very long ( between 15s and 40s per host ).

Reply:
Ah, ok. Are the also other ports we should test? Because 22 would require the Linux host to run an ssh server which may not always be the case.

Reply:
My checks are now:
- Linbo: only 2222 is open
- Linux: 22 is open or filtered and 135 not open
- Windows: 135 is open and 22 is not open